web_monitoring/diffing_server.py

```diff
@@ -67,17 +69,47 @@ def get(self, differ):
         try:
             func = self.differs[differ]
         except KeyError:
-            self.send_error(404)
+            self.send_error(404, reason="Diffing method not found.")
             return
 
         # If params repeat, take last one. Decode bytes into unicode strings.
         query_params = {k: v[-1].decode() for k, v in
                         self.request.arguments.items()}
-        a = query_params.pop('a')
-        b = query_params.pop('b')
-
+        try:
+            a = query_params.pop('a')
+            b = query_params.pop('b')
+        except KeyError:
+            self.send_error(
+                        400, reason="Malformed request. A URL parameter is missing.")
+            return
         # Fetch server response for URLs a and b.
-        res_a, res_b = yield [client.fetch(a), client.fetch(b)]
+        res_a, res_b = yield [client.fetch(a, raise_error=False), client.fetch(b, raise_error=False)]
+        
+        #Check if the HTTP requests were successfull and handle exeptions
+        if res_a.error is not None:
+            try:
+                res_a.rethrow()
+            except (ValueError, IOError):
+                if res_a.code == 599:
+                    self.send_error(
+                        400, reason=str(res_a.error))
+                else:
+                    self.send_error(
+                        res_a.code, reason=str(res_a.error))
+                return
+
+        if res_b.error is not None:
```

Comment:
These two error handling blocks (for `res_a` and `res_b`) are exactly the same, so it would be btter to encapsulate this error logic into a function that you can call once for each response.